core/services/chainlink/config.go

```diff
@@ -180,7 +220,8 @@ func (s *Secrets) setEnv() error {
 		}
 	}
 	if env.DatabaseAllowSimplePasswords.IsTrue() {
-		s.Database.AllowSimplePasswords = true
+		s.setDefaults()
```

Comment:
Is this right?

Reply:
The intent is that if `CL_DATABASE_ALLOW_SIMPLE_PASSWORDS` is explicitly set to true it will be honored. SetDefaults() is needed in case s.Database.AllowSimplePasswords is `nil`. 

Does something smell off here?

Reply:
I see now that the _implementation_ of `setDefaults` only has to do with `AllowSimplePasswords`, but the method name looks general, and is used generally from elsewhere. Would it make sense to name the method for this particular field? Alternatively, could we call this always, instead of conditionally?

Reply:
Hmm, yea, good call. It's not clear from reading what it handles. 

We talked a bit about this during some earlier passes https://github.com/smartcontractkit/chainlink/pull/9590#discussion_r1229986792, but I agree there's an opportunity to improve readability here :)